src/store/shopping_cart/shoppingCart_slice.ts

```diff
@@ -33,11 +34,56 @@ const shoppingCard_slice = createSlice({
       } else {
         const copyCardItems: unknown = [
           ...state.products,
-          { ...product, quantity: 1 },
+          { ...product, quantity: 1, totalPrice: product?.price },
         ];
         state.products = copyCardItems as IShoppingCard[];
       }
+      if (productIndex >= 0) {

```

Comment:
"cardTotalAmount is of type number.
Currently, we have logic that assigns a value of type string to the cardTotalAmount variable.
It would be better to refactor the logic in these reducers, as it is overly complicated and missed types.
I've create a [PR](https://github.com/mskmee/gadget-house/pull/41), for changes example